src/services/htmlCompletion.ts

```diff
@@ -175,9 +175,41 @@ export function doComplete(document: TextDocument, position: Position, htmlDocum
 				});
 			});
 		});
+		collectDataAttributesSuggestions(range);
 		return result;
 	}
 
+	function collectDataAttributesSuggestions(range: Range) {
+
+		const dataAttr = 'data-';
+		let dataAttributes: string[] = [];
+
+		function addNodeDataAttributes(node: Node) {
+			node.attributeNames.filter(attr => attr.substr(0, dataAttr.length) === dataAttr).forEach(attr => {
+				if (dataAttributes.indexOf(attr) === -1) {
```

Comment:
Use a object or Map to check for extistece
```ts
if (!dataAttributes['attr']) {
    dataAttributes['attr'] = true;
    ...
```

Reply:
ok I'll make the modification ASAP